lib/library.cpp

```diff
@@ -216,11 +216,13 @@ Library::Error Library::load(const tinyxml2::XMLDocument &doc)
             int allocationId = 0;
             for (const tinyxml2::XMLElement *memorynode = node->FirstChildElement(); memorynode; memorynode = memorynode->NextSiblingElement()) {
                 if (strcmp(memorynode->Name(),"dealloc")==0) {
-                    const std::map<std::string, AllocFunc>::const_iterator it = mDealloc.find(memorynode->GetText());
-                    if (it != mDealloc.end()) {
-                        allocationId = it->second.groupId;
-                        break;
+                    const auto names = getnames(memorynode->GetText());
+                    for (const auto& n : names) {
+                        const std::map<std::string, AllocFunc>::const_iterator it = mDealloc.find(n);
+                        if (it != mDealloc.end())
+                            allocationId = it->second.groupId;
                     }
+                    break;
```

Comment:
hmm we used to handle multiple "dealloc" elements. and break when we found a match in mDealloc.
Now we only handle the first "dealloc"..
I don't understand why we would handle several "dealloc" elements like this to start with.. but was this change of behavior intended?

Reply:
It was not intended, restored old behavior.